setup/src/installer.ts

```diff
@@ -220,7 +226,37 @@ async function ghcup(tool: Tool, version: string, os: OS): Promise<void> {
   if (returnCode === 0) await exec(bin, ['set', tool, version]);
 }
 
-function getChocoPath(tool: Tool, version: string): string {
+async function getChocoPath(
+  tool: Tool,
+  version: string
+): Promise<string | null> {
+  if (!process.env.ChocolateyInstall) {
+    return null;
+  }
+
+  if (tool === 'cabal') {
+    const chocoToolPath = join(
+      `${process.env.ChocolateyInstall}`,
+      'lib',
+      `${tool}.${version}`
+    );
+
+    const globber = await glob.create(`${chocoToolPath}/tools/*/cabal.exe`);
+
+    for await (const file of globber.globGenerator()) {
+      const chocoToolBinPath = file
```

Comment:
Any reason to prefer this chain over `path.dirname(file)`?

Reply:
Fixed